src/librustdoc/html/static/js/search.js

```diff
@@ -1698,10 +1749,24 @@ function initSearch(rawSearchIndex) {
             }
 
             // If the result is too "bad", we return false and it ends this search.
-            if (!unifyFunctionTypes(row.type.inputs, parsedQuery.elems)) {
+            let mgens;
```

Comment:
The `handleArgs` function became super huge. Would it be possible to move the callbacks you create in it outside as functions (with more args)? Or if you see another way to reduce its size, please do.